src/components/views/spaces/QuickThemeSwitcher.tsx

```diff
@@ -0,0 +1,93 @@
+/*
+Copyright 2022 The Matrix.org Foundation C.I.C.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import React, { useMemo } from "react";
+
+import { _t } from "../../../languageHandler";
+import { Action } from "../../../dispatcher/actions";
+import { findNonHighContrastTheme, getOrderedThemes } from "../../../theme";
+import Dropdown from "../elements/Dropdown";
+import ThemeChoicePanel from "../settings/ThemeChoicePanel";
+import SettingsStore from "../../../settings/SettingsStore";
+import { SettingLevel } from "../../../settings/SettingLevel";
+import dis from "../../../dispatcher/dispatcher";
+import { RecheckThemePayload } from "../../../dispatcher/payloads/RecheckThemePayload";
+import PosthogTrackers from "../../../PosthogTrackers";
+
+type Props = {
+    requestClose: () => void;
+};
+
+const MATCH_SYSTEM_THEME_ID = 'MATCH_SYSTEM_THEME_ID';
+
+const QuickThemeSwitcher: React.FC<Props> = ({ requestClose }) => {
+    const orderedThemes = useMemo(getOrderedThemes, []);
+
+    const themeState = ThemeChoicePanel.calculateThemeState();
+    const nonHighContrast = findNonHighContrastTheme(themeState.theme);
+    const theme = nonHighContrast ? nonHighContrast : themeState.theme;
+    const { useSystemTheme } = themeState;
+
+    const themeOptions = [{
+        id: MATCH_SYSTEM_THEME_ID, name: 'Match system',
+    }, ...orderedThemes];
```

Comment:
Also, adding this option will require design/product sign off, otherwise it should behave like the theme toggle in the User Menu which just disables match-system as soon as you toggle it

Reply:
Also *NEEDS* i18n